src/core/state.py

```diff
@@ -44,19 +44,84 @@ def __init__(self, state_dict: Dict = {}):
         self.propulsion_on = False
         self.solenoid_actuation_on = False
 
+        self.update(state_dict)
+
+    def update(self, state_dict: Dict[str, Union[int, float, bool]]) -> None:
         for key, value in state_dict.items():
             if key in self.__dict__.keys():
                 setattr(self, key, value)
 
     def to_array(self):
         """
-        to_array() is the representation of the values of the fields as an array.
+        to_array() is the representation of the values of the fields as an
+         array.
 
         Returns:
             Numpy array: contains all values stored in the fields.
         """
         return np.array(list(self.__dict__.values()))
 
+    def from_array(self, state_array: np.ndarray):
+
+        new_state = dict(zip(self.__dict__.keys(), state_array))
+        self.update(new_state)
+
+    def __eq__(self, other):
+        """
+
+        Args:
+            other (State): the "other" object self is being compared to.
+
+        Returns:
+            True iff other is a State object with equal attributes.
+        """
+        if type(other) == State:
+            return self.__dict__ == other.__dict__
+        return False
+
+
+STATE_ARRAY_ORDER = list(State().__dict__.keys())
+
+
+def array_to_state(values: np.ndarray) -> State:
+    """Converts a numpy array or list into a `State` object.
+     This assumes that the items in `state_array` are consistent with
+     `STATE_ARRAY_ORDER`(which is an assumption that will probably lead
+     to many bugs in the future...)
+
+    Args:
+        state_array (np.ndarray): n-by-1 numpy array of each state
+    """
+    return State(dict(zip(STATE_ARRAY_ORDER, values)))
+
+
+@dataclass
+class StateTime:
```

Comment:
Why did we decide to decouple state and time into separate classes?

Reply:
Because time is not really an "evolved" state of the system. Scipy's integrator makes a distinction between time and state. I might change my mind later on this, but here we are now.